Samples.Swift/Samples.Swift/ViewControllers/CheckoutViewController/CheckoutViewController.swift

```diff
@@ -0,0 +1,478 @@
+    paypalButton.heightAnchor.constraint(equalToConstant: 40).isActive = true
+  }
+
+  @objc
+  func checkoutFlowOptionChanged(_ sender: UISegmentedControl) {
+    switch sender.selectedSegmentIndex {
+    case 0:
+      checkoutButton.isHidden = false
+      paypalButton.isHidden = true
+      checkoutButton.setTitle("Checkout with Order", for: .normal)
+
+    case 1:
+      checkoutButton.isHidden = false
+      paypalButton.isHidden = true
+      checkoutButton.setTitle("Checkout with ECToken", for: .normal)
+
+    case 2:
+      checkoutButton.isHidden = true
+      paypalButton.isHidden = false
+
+    default:
+      break
+    }
+  }
+
+  private func processOrderActions(with approval: Approval) {
+    switch approval.data.intent {
+    case "AUTHORIZE":
+      approval.actions.authorize { success, error in
+        if let error = error {
+          self.errorResultAction(errorMessage: error.localizedDescription)
+        }
+        else if let success = success {
+          self.successResultAction(orderData: success.data.orderData, orderId: success.data.id)
+        }
+        else {
+          print("Unhandled State: No Error and No Success Response")
+        }
+      }
+
+    case "CAPTURE", "SALE":
+      approval.actions.capture { success, error in
+        if let error = error {
+          self.errorResultAction(errorMessage: error.localizedDescription)
+        }
+        else if let success = success {
+          self.successResultAction(orderData: success.data.orderData, orderId: success.data.id)
+        }
+        else {
+          print("Unhandled State: No Error and No Success Response")
+        }
+      }
+
+    default:
+      break
+    }
+  }
+
+  func successResultAction(orderData: [String: Any], orderId: String) {
+    print("Successful Response Data: \(orderData)")
+
+    let resultViewController = ResultViewController()
+    resultViewController.resultLabel.text = "Success 😎"
+    resultViewController.messageLabel.text = "Your order number is \(orderId)"
+    resultViewController.modalPresentationStyle = .fullScreen
+
+    self.present(resultViewController, animated: true)
+  }
+
+  func errorResultAction(errorMessage: String) {
+      print("Capture Action Error Response: \(errorMessage)")
+
+      let resultViewController = ResultViewController()
+      resultViewController.resultLabel.text = "Error 😕"
+      resultViewController.messageLabel.text = "Your order failed with \(errorMessage)"
+      resultViewController.modalPresentationStyle = .fullScreen
+
+    self.present(resultViewController, animated: true)
+  }
+
+  func getItemTotal() -> String {
+    var total: Double = 0
+
+    for item in items {
+      guard let unitPrice = Double(item.unitAmount.value) else { return "" }
+      guard let quantity = Double(item.quantity) else { return "" }
+
+      let itemTotalPrice = unitPrice * quantity
+      total += itemTotalPrice
+    }
```

Comment:
This block can be replaced by  : 
```swift
    let total = items.reduce(0) { (result, item) -> Double in
      let unitAmount = Double(item.unitAmount.value) ?? 0
      let quantity = Double(item.quantity)) ?? 0
      result += quantity * unitAmount
    }
```

This would also prevent us from returning out early if one item was without a unit amount / quantity. Mostly just a fun exercise in making this a bit more swifty.

Reply:
the correct block for this is actually 
```swift
    let total = items.reduce(0) { (runningTotal, item) -> Double in
      let unitAmount = Double(item.unitAmount.value) ?? 0
      let quantity = Double(item.quantity) ?? 0
      
      return runningTotal + (quantity * unitAmount)
    }
```
my bad! :